compiler/rustc_passes/src/dead.rs

```diff
@@ -440,15 +422,31 @@ impl<'v, 'k, 'tcx> ItemLikeVisitor<'v> for LifeSeeder<'k, 'tcx> {
         }
     }
 
-    fn visit_trait_item(&mut self, _item: &hir::TraitItem<'_>) {
-        // ignore: we are handling this in `visit_item` above
+    fn visit_trait_item(&mut self, trait_item: &hir::TraitItem<'_>) {
```

Comment:
is there a reason you did this move for trait items and not for impl items?

Reply:
The treatment of impl items requires knowing whether they are implementing a trait. This information is not accessible from the impl item itself, so I could do the move.